File: mytrial-fe/src/service/DonorService.js
// SJSU CMPE 138 Fall 2021 TEAM1
import axios from "axios";
const MYTRIAL_API = 'http://127.0.0.1:8080';
const ADD_DONOR_API_BASE_URL = `${MYTRIAL_API}/add_donor/`;
const UPDATE_DONOR_API_BASE_URL = `${MYTRIAL_API}/update_donor/`;
const DONOR_API_BASE_URL = `${MYTRIAL_API}/donor/`;

class DonorService {
    CreateDonor(donor, config){
        return axios.post(ADD_DONOR_API_BASE_URL, donor, config);
    }
    UpdateDonor(patient_id, donor, config){
        return axios.post(UPDATE_DONOR_API_BASE_URL + patient_id, donor, config);
    }

    getDonorByPatientId(patient_id){
        return axios.get(DONOR_API_BASE_URL + patient_id);
    }
}
export default new DonorService()

File: myTrial-api/src/main/java/com/cmpe138/mytrial/service/DonorService.java
//SJSU CMPE 138 Fall 2021 TEAM1
package com.cmpe138.mytrial.service;

import com.cmpe138.mytrial.model.Donor;

public interface DonorService {
    void createDonor(String patient_id, String name, String email);
    void updateDonorByPatientId(String patient_id, String name, String email);
    Donor getDonorByPatientId(String patient_id);
}
